bytecounterprocessor/bytecounterProcessor.go

```diff
@@ -0,0 +1,44 @@
+package bytecounterprocessor
+
+import (
+	"context"
+
+	"github.com/honeycombio/opentelemetry-collector-configs/bytecounterprocessor/internal/metadata"
+	"go.opentelemetry.io/collector/pdata/plog"
+	"go.opentelemetry.io/collector/pdata/pmetric"
+	"go.opentelemetry.io/collector/pdata/ptrace"
+	"go.opentelemetry.io/collector/processor"
+)
+
+type byteCounterProcessor struct {
+	telemetryBuilder *metadata.TelemetryBuilder
+}
+
+func newByteCountProcessor(settings processor.Settings) (*byteCounterProcessor, error) {
+	telemetryBuilder, err := metadata.NewTelemetryBuilder(settings.TelemetrySettings)
+	if err != nil {
+		return nil, err
+	}
+
+	return &byteCounterProcessor{
+		telemetryBuilder: telemetryBuilder,
+	}, nil
+}
+
+func (p *byteCounterProcessor) processTraces(ctx context.Context, tracesData ptrace.Traces) (ptrace.Traces, error) {
+	m := ptrace.ProtoMarshaler{}
+	p.telemetryBuilder.TracesBytesCount.Add(ctx, int64(m.TracesSize(tracesData)))
```

Comment:
Does this measure compressed or uncompressed?

Reply:
It walks down the [proto definitions](https://github.com/open-telemetry/opentelemetry-collector/blob/pdata/v1.26.0/pdata/internal/data/protogen/trace/v1/trace.pb.go#L1432) so I believe is the uncompressed size.